src/content-script/index.mjs

```diff
@@ -28,11 +21,40 @@ async function run(question) {
   port.postMessage({ question });
 }
 
-const searchInput = document.getElementsByName("q")[0];
-if (searchInput && searchInput.value) {
-  // only run on first page
-  const startParam = new URL(location.href).searchParams.get("start") || "0";
-  if (startParam === "0") {
+if(location.hostname.match("startpage")){
+  // startpage
+  document.querySelector(".layout-web__sidebar.layout-web__sidebar--web").appendChild(container);
+  const searchInput = document.getElementById("q");
+  if (searchInput && searchInput.value) {
     run(searchInput.value);
   }
+}else if(location.hostname.match("duckduckgo")){
+  // duckduckgo
+  document.getElementById("web_content_wrapper").prepend(container);
+  const searchInput = document.getElementsByName("q")[0];
+  if (searchInput && searchInput.value) {
+    // only run on first page
+    const startParam = new URL(location.href).searchParams.get("start") || "0";
+    if (startParam === "0") {
+      run(searchInput.value);
+    }
+  }
+}else if(location.hostname.match("google")){
+  // google
+  const siderbarContainer = document.getElementById("rhs");
+  if (siderbarContainer) {
+    siderbarContainer.prepend(container);
+  } else {
+    container.classList.add("sidebar-free");
+    document.getElementById("rcnt").appendChild(container);
+  }
+  const searchInput = document.getElementsByName("q")[0];
+  if (searchInput && searchInput.value) {
+    // only run on first page
+    const startParam = new URL(location.href).searchParams.get("start") || "0";
+    if (startParam === "0") {
+      run(searchInput.value);
+    }
+  }
```

Comment:
Maybe refactor this into a function so that other search engines can be cleanly added